### PR TITLE
test: close coverage gaps in simple_packing, zfp, and CLI

### PR DESCRIPTION
**Pull request by tlmquintino**
## Summary

Addresses the TODO item for code coverage by adding 23 targeted Rust tests for the specific uncovered paths identified by `cargo-llvm-cov`.

### Coverage Improvements

| File | Before | After | Tests Added |
|------|--------|-------|-------------|
| `simple_packing.rs` | 94.0% | **96.8%** | 17 — NaN detection, empty/zero-bit encode/decode, insufficient data, bpv validation, write/read bits edge cases |
| `zfp_ffi.rs` | 88.1% | **92.5%** | 5 — empty compress/decompress, range exceeds total, accuracy/precision mode round-trips |
| `info.rs` (CLI) | 0% | **95.9%** | 3 — single file, empty file, missing file |

### What remains (diminishing returns)
- CLI dump/ls/get/reshuffle: 0% — print-to-stdout, tested via C++ integration
- FFI: 0% — 1844 lines tested by 109 C++ tests, not measurable by cargo-llvm-cov
- Core metadata.rs: 89% — remaining gaps are malformed-CBOR error paths

### Stats
- 313 Rust tests (was 290), 911 total across all languages
- 0 clippy warnings

Docs Preview
https://sites.ecmwf.int/docs/dev-section/tensogram/pull-requests/PR-14

### CODE REVIEW

File: crates/tensogram-cli/src/commands/info.rs [hunk 1]
```diff
@@ -22,3 +22,57 @@ pub fn run(files: &[PathBuf]) -> Result<(), Box<dyn std::error::Error>> {
...
+    #[test]
+    fn info_empty_file() {
+        let dir = tempfile::tempdir().unwrap();
+        let path = make_test_file(dir.path(), 0);
+        run(&[path]).unwrap();
+    }
+
+    #[test]
+    fn info_missing_file() {
+        let result = run(&[PathBuf::from("/nonexistent.tgm")]);
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply crates/tensogram-cli/src/commands/info.rs]: Fixed in 2fd18b28.

File: plans/TODO.md [hunk 2]
```diff
@@ -42,5 +42,5 @@ For speculative ideas, see `IDEAS.md`.
 
 ## Code Quality
 
-- [~] code coverage: Rust core ~91%, Python 200 tests, C++ 109 tests. Remaining gaps are szip/zfp edge cases and CLI/FFI (tested via C++/integration, not measurable by cargo-llvm-cov). Total 888 tests.
+- [x] ~~code coverage~~ → simple_packing 97%, zfp 92%, CLI info 96%, 23 new Rust tests (313 total). Remaining: CLI dump/ls/get/reshuffle at 0% (integration-tested via binary), FFI at 0% (109 C++ tests). Total 911 tests.
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply plans/TODO.md]: Fixed in 2fd18b28.

File: crates/tensogram-cli/src/commands/convert_grib.rs [hunk 3]
```diff
@@ -55,3 +55,167 @@ pub fn run(
...
+    fn convert_to_stdout() {
+        // No output file → writes to stdout (just verify it doesn't crash)
+        run(
+            &[testdata("2t.grib2")],
+            None, // stdout
+            false,
+            false,
+        )
+        .unwrap();
+    }
```
[the remark this answers is not in the document]
Comment (tlmquintino) [reply crates/tensogram-cli/src/commands/convert_grib.rs]: Fixed in 2fd18b28.
